src/annotator/anchoring/xpath.js

```diff
@@ -1,52 +1,113 @@
-import { findChild } from './xpath-util';
-
 export { xpathFromNode } from './xpath-util';
 
 /**
- * Finds an element node using an XPath relative to the document root.
+ * Return the `index`'th immediate child of `element` whose tag name is
+ * `nodeName` (case insensitive).
+ *
+ * @param {Element} element
+ * @param {string} nodeName
+ * @param {number} index
  */
-function nodeFromXPathFallback(xpath, root) {
-  const steps = xpath.substring(1).split('/');
-  let node = root;
-  steps.forEach(step => {
-    let [name, idx] = step.split('[');
-    idx = idx ? parseInt(idx.split(']')[0]) : 1;
-    node = findChild(node, name.toLowerCase(), idx);
-  });
-  return node;
+function nthChildOfType(element, nodeName, index) {
+  nodeName = nodeName.toUpperCase();
+
+  let matchIndex = -1;
```

Comment:
`let found` started at `0`, and then it first compared at 1. this starts at -1 and first compares at 0. I guess I'm confused why they are different.

Reply:
In `findChild`, `index` was a 1-based index and `found` is the count of matching elements found. In `nthChildOfType` the `index` param is a 0-based index. The reason for using 1-based indexes previously is probably because XPaths use 1-based indexing (eg. `/div[1]` for the first `div` child). The code that calls `nthChildOfType` has been adjusted to convert the 1-based index from the XPath to an 0-based index during parsing.